### PR TITLE
Migration: deprecate borderless-accounts endpoint, and provide account-detail and balance-statement endpoints.

### LINKED ISSUE #24: Deprecation notice: borderless-accounts endpoints
The `borderless-accounts` API endpoints are deprecated, so an alternative mechanism should be provided to access account details - likely the [Account Details API](https://docs.wise.com/api-docs/api-reference/bank-account-details).

Originated-from emfcamp/Website#1328

### PR DESCRIPTION
**Pull request by jayaddison**
It's likely going to make sense to introduce some datatypes (`dataclasses` perhaps?) for the result objects in the nearish future; for now though, make the endpoints available.

My understanding is that the `DeprecationWarning` is unlikely to appear in many downstream applications at runtime, since it's [ignored by default](https://docs.python.org/3/library/exceptions.html#DeprecationWarning); it's most likely to appear when test suites run.

Resolves #24.

### CODE REVIEW

**Comment by marksteward** [comment]
Looks good from a read! This still requires SCA, is that done differently to the old endpoints, or are we still going to need to find an alternative to private keys? We obviously need to make this change, but I'm just wondering if this takes away the potential imminent doom from #24.

**Comment by jayaddison** [comment]
> Looks good from a read! This still requires SCA, is that done differently to the old endpoints, or are we still going to need to find an alternative to private keys? We obviously need to make this change, but I'm just wondering if this takes away the potential imminent doom from #24.

Thanks!  My sense - and I'll check again tomorrow because I already had to remind myself - is that for business accounts, retrieving statements using signature-based authentication is still OK.  That's not something I've confirmed with Wise, though.

File: pywisetransfer/deprecation.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+def deprecated(message, *args, **kwargs):
+    """Decorator to indicate that a method is deprecated, with an optional
+    message to emit in the warning.
+
+    Written following guidance from:
+    https://blog.miguelgrinberg.com/post/the-ultimate-guide-to-python-decorators-part-iii-decorators-with-arguments
+    """
+
+    # General-case: handle decorators with arguments; @deprecated(message='foo')
+    def decorator(f):
```
Comment (jayaddison) [inline pywisetransfer/deprecation.py:15]: @marksteward not important if you're busy, but are you familiar enough with Python decorator magic to help me avoid the object signatures for items decorated as `@deprecated` from looking.. spammy?

```pyi
>>> import pywisetransfer
>>> client = pywisetransfer.Client(api_key='testing')
>>> client.borderless_accounts.statement
<bound method deprecated.<locals>.decorator.<locals>.deprecation_notice of <pywisetransfer.borderless_account.BorderlessAccount object at <0xblah>>
```

Comment (jayaddison) [reply pywisetransfer/deprecation.py]: (please ignore: this has been resolved)

**Pull request merged**